// carbon-components-react/es/components/Checkbox/Checkbox-story.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, text } from '@storybook/addon-knobs';
import Checkbox from '../Checkbox';
import CheckboxSkeleton from '../Checkbox/Checkbox.Skeleton';
import { settings } from 'carbon-components';
var prefix = settings.prefix;

var props = function props() {
  return {
    className: 'some-class',
    labelText: text('Label text (labelText)', 'Checkbox label'),
    indeterminate: boolean('Intermediate (indeterminate)', false),
    disabled: boolean('Disabled (disabled)', false),
    hideLabel: boolean('No label (hideLabel)', false),
    wrapperClassName: text('Wrapper CSS class name (wrapperClassName)', ''),
    onChange: action('onChange')
  };
};

export default {
  title: 'Checkbox',
  decorators: [withKnobs],
  parameters: {
    component: Checkbox,
    subcomponents: {
      CheckboxSkeleton: CheckboxSkeleton
    }
  }
};
export var Checked = function Checked() {
  var checkboxProps = props();
  return /*#__PURE__*/React.createElement("fieldset", {
    className: "".concat(prefix, "--fieldset")
  }, /*#__PURE__*/React.createElement("legend", {
    className: "".concat(prefix, "--label")
  }, "Checkbox heading"), /*#__PURE__*/React.createElement(Checkbox, _extends({
    defaultChecked: true
  }, checkboxProps, {
    id: "checkbox-label-1"
  })), /*#__PURE__*/React.createElement(Checkbox, _extends({
    defaultChecked: true
  }, checkboxProps, {
    id: "checkbox-label-2"
  })));
};
Checked.storyName = 'checked';
Checked.parameters = {
  info: {
    text: "\n    Checkboxes are used when there is a list of options and the user may select multiple options, including all or none.\n    The example below shows how the Checkbox component can be used as an uncontrolled component that is initially checked\n    by setting the defaultChecked property to true. To use the component in a controlled way, you should set the\n    checked property instead.\n  "
  }
};
export var Unchecked = function Unchecked() {
  var checkboxProps = props();
  return /*#__PURE__*/React.createElement("fieldset", {
    className: "".concat(prefix, "--fieldset")
  }, /*#__PURE__*/React.createElement("legend", {
    className: "".concat(prefix, "--label")
  }, "Checkbox heading"), /*#__PURE__*/React.createElement(Checkbox, _extends({}, checkboxProps, {
    id: "checkbox-label-1"
  })), /*#__PURE__*/React.createElement(Checkbox, _extends({}, checkboxProps, {
    id: "checkbox-label-2"
  })));
};
Unchecked.storyName = 'unchecked';
Unchecked.parameters = {
  info: {
    text: "\n      Checkboxes are used when there is a list of options and the user may select multiple options, including all or none.\n      The example below shows how the Checkbox component can be used as an uncontrolled component that is initially\n      unchecked. To use the component in a controlled way, you should set the checked property instead.\n    "
  }
};
export var Skeleton = function Skeleton() {
  return /*#__PURE__*/React.createElement("div", {
    "aria-label": "loading checkbox",
    "aria-live": "assertive",
    role: "status",
    tabIndex: "0" // eslint-disable-line jsx-a11y/no-noninteractive-tabindex

  }, /*#__PURE__*/React.createElement(CheckboxSkeleton, null));
};
Skeleton.storyName = 'skeleton';
Skeleton.parameters = {
  info: {
    text: "\n      Placeholder skeleton state to use when content is loading.\n    "
  }
};
